Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -214,24 +213,32 @@ IRelocator relocator
       }
     }
 
-    private async Task<DafnyDocument> LoadVerificationTasksAsync(Task<DafnyDocument> resolvedDocumentTask, CancellationToken cancellationToken) {
+    public async Task<DafnyDocument> LoadVerificationTasksAsync(Task<DafnyDocument> resolvedDocumentTask, CancellationToken cancellationToken) {
 #pragma warning disable VSTHRD003
       var resolvedDocument = await resolvedDocumentTask;
 #pragma warning restore VSTHRD003
-      var withVerificationTasks = await documentLoader.PrepareVerificationTasksAsync(resolvedDocument, cancellationToken);
-      if (resolvedDocument.ImplementationViews == null) {
-        return withVerificationTasks;
-      }
+      try {
+        var withVerificationTasks =
+          await documentLoader.PrepareVerificationTasksAsync(resolvedDocument, cancellationToken);
+        if (resolvedDocument.ImplementationViews == null) {
+          return withVerificationTasks;
+        }
 
-      return withVerificationTasks with {
-        ImplementationViews = withVerificationTasks.ImplementationViews!.ToDictionary(
-          kv => kv.Key,
-          kv =>
-            kv.Value with {
-              Diagnostics = resolvedDocument.ImplementationViews.GetValueOrDefault(kv.Key)?.Diagnostics ?? kv.Value.Diagnostics
-            }
-        ),
-      };
+        return withVerificationTasks with {
+          ImplementationViews = withVerificationTasks.ImplementationViews!.ToDictionary(
+            kv => kv.Key,
+            kv =>
+              kv.Value with {
+                Diagnostics = resolvedDocument.ImplementationViews.GetValueOrDefault(kv.Key)?.Diagnostics ??
+                              kv.Value.Diagnostics
+              }
+          ),
+        };
+      } catch (Exception e) {
```

Comment:
Since this is an `async` method, any exceptions are already caught in the `Task`. I think it might be better to handle those exceptions at a higher level, where the task is handled.

Where we call `translatedDocument.ToObservableSkipCancelled()`, any exception in the Task is translated to an OnError event in the observable, and that may subsequently block all further notifications, since the `Subject` inside `MergeOrdered` will, like any `Subject`, stop sending OnNext events after it has sent an OnError event. Let's fix that!

Reply:
> Where we call `translatedDocument.ToObservableSkipCancelled()`, any exception in the Task

I don't think it's true for "any exception", only "OperationCancelledException".

Reply:
I'm suggesting to do this: https://github.com/dafny-lang/dafny/pull/2254/commits/24d6f13d6a3a9d08c577771d1b87dedd053f0c42

Without that change, any exception other than `OperationCancelledException `, that occurs in Tasks on which ToObservableSkipCancelled is called, will corrupt the language server so it doesn't send notifications any more.

Reply:
I like this commit, that makes sense.